tests/run/legacy_implicit_noexcept.pyx

```diff
@@ -0,0 +1,39 @@
+# cython: legacy_implicit_noexcept=True
+# mode: run
+# tag: warnings
+
+cdef int func_implicit(int a, int b):
+    raise RuntimeError
+
+cdef int func_noexcept(int a, int b) noexcept:
+    raise RuntimeError
+
+cdef int func_star(int a, int b) except *:
+    raise RuntimeError
+
```

Comment:
A few possibly missing tests:

* `except -1` or similar test? I think it should work, but it's an obvious omission.
* return `object` (or don't specify the return). Given that this behaviour remains the same it possibly shouldn't be a warning.
* function pointers? I think they remain the same?
* `@cython.cfunc` - not quite sure what they do but it should probably be tested

Reply:
How about external functions? But in this case we need to refactor test as .srctree I suppose...

Reply:
Tests implemented for all @da-woods points.